llmfoundry/data/packing.py

```diff
@@ -86,24 +102,28 @@ def pack(self, batch: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
             ]
         # Cut everything down to size
         sizes, trimmed_examples = _trim_batch(batch)
-        return self._pack_trimmed_examples(trimmed_examples, sizes)
+        packed_batch = self._pack_trimmed_examples(trimmed_examples, sizes)
+        assert packed_batch is not None
```

Comment:
use ValueError over assert
```suggestion
        if packed_batch is None:
            raise ValueError(...)
```

Reply:
This is purely for pyright, not actual error handling

Reply:
also can you add 
```
if self._is_profiling:
    raise ValueError('`pack` should not be called while profiling. Please use `_pack_trimmed_examples`.')
```

Reply:
`pack` is never called when profiling

Reply:
So that someone doesn't call `pack` when creating a BinPackCollator in profiling mode